go/tools/builders/stdliblist.go

```diff
@@ -145,6 +149,35 @@ func flatPackageForStd(execRoot string, pkg *goListPackage) *flatPackage {
 	return newPkg
 }
 
+// In Go 1.18, the standard library started using go:embed directives.
+// When Bazel runs this action, it does so inside a sandbox where GOROOT points
+// to an external/go_sdk directory that contains a symlink farm of all files in
+// the Go SDK.
+// If we run "go list" with that GOROOT, this action will fail because those
+// go:embed directives will refuse to include the symlinks in the sandbox.
+//
+// To work around this, cloneGoRoot creates a copy of external/go_sdk into a new
+// cloneBase directory while retaining its path relative to the root directory.
+// So "$OUTPUT_BASE/external/go_sdk" becomes
+// {cloneBase}/external/go_sdk", which will be set at GOROOT later.
+// This ensures that file paths in the generated JSON are still valid.
+//
+// cloneGoRoot returns the new GOROOT we should run
+// under.
+func cloneGoRoot(execRoot, relativeGoroot, cloneBase string) (newGoRoot string, err error) {
+	goroot := filepath.Join(execRoot, relativeGoroot)
+	newGoRoot = filepath.Join(cloneBase, relativeGoroot)
+	if err := os.MkdirAll(newGoRoot, 01755); err != nil {
+		return "", err
+	}
+
+	if err := replicate(goroot, newGoRoot, replicatePaths("src", "pkg/tool", "pkg/include")); err != nil {
```

Comment:
"pkg/tool" is needed because go list would fail without it
```
Use --sandbox_debug to see verbose messages from the sandbox
go: no such tool "compile"
go: no such tool "compile"
go: no such tool "compile"
go: no such tool "compile"
go: no such tool "compile"
stdliblist: error running subcommand external/go_sdk/bin/go: exit status 2
```
maybe it's used in internal packages [such as cfg](https://github.com/golang/go/blob/go1.18/src/cmd/go/internal/cfg/cfg.go#L397) which is then used in go commands

Reply:
Good finding. Yes, `go list` needs to call `compile`. Can you add comments to explain this finding?

Reply:
done